openmc/filter.py

```diff
@@ -732,37 +732,40 @@ def get_bin_index(self, filter_bin):
         # Filter bins for a mesh are an (x,y,z) tuple. Convert (x,y,z) to a
         # single bin -- this is similar to subroutine mesh_indices_to_bin in
         # openmc/src/mesh.F90.
-        if len(self.mesh.dimension) == 3:
+        n_dim = len(self.mesh.dimension)
+        if n_dim == 3:
+            i, j, k = filter_bin
             nx, ny, nz = self.mesh.dimension
-            val = (filter_bin[0] - 1) * ny * nz + \
-                  (filter_bin[1] - 1) * nz + \
-                  (filter_bin[2] - 1)
-        else:
+            return (i - 1) + (j - 1)*nx + (k - 1)*nx*ny
+        elif n_dim == 2:
+            i, j, *_ = filter_bin
             nx, ny = self.mesh.dimension
-            val = (filter_bin[0] - 1) * ny + \
-                  (filter_bin[1] - 1)
-
-        return val
+            return (i - 1) + (j - 1)*nx
+        else:
+            return filter_bin[0] - 1
 
     def get_bin(self, bin_index):
         cv.check_type('bin_index', bin_index, Integral)
         cv.check_greater_than('bin_index', bin_index, 0, equality=True)
         cv.check_less_than('bin_index', bin_index, self.num_bins)
 
-        # Construct 3-tuple of x,y,z cell indices for a 3D mesh
-        if len(self.mesh.dimension) == 3:
+        n_dim = len(self.mesh.dimension)
+        if n_dim == 3:
+            # Construct 3-tuple of x,y,z cell indices for a 3D mesh
             nx, ny, nz = self.mesh.dimension
-            x = bin_index / (ny * nz)
-            y = (bin_index - (x * ny * nz)) / nz
-            z = bin_index - (x * ny * nz) - (y * nz)
+            x = bin_index % nx + 1
+            y = (bin_index % (nx * ny)) // nx + 1
+            z = bin_index // (nx * ny) + 1
             return (x, y, z)
 
-        # Construct 2-tuple of x,y cell indices for a 2D mesh
-        else:
+        elif n_dim == 2:
+            # Construct 2-tuple of x,y cell indices for a 2D mesh
             nx, ny = self.mesh.dimension
-            x = bin_index / ny
-            y = bin_index - (x * ny)
+            x = bin_index % nx + 1
```

Comment:
Put the modulo terms in parentheses for clarity.

Reply:
Good idea